### PR TITLE
Implement metrics/tracing on top of OpenTelemetry

### PR DESCRIPTION
**Pull request by Pliner**
Instead of a self-made metrics facade, using OpenTelemetry as a defacto standard for metrics/tracing might be better.

### CODE REVIEW

File: aio_request/__init__.py [hunk 1]
```diff
@@ -135,7 +126,8 @@
 try:
     import prometheus_client
 
-    from .prometheus import PROMETHEUS_METRICS_PROVIDER, PrometheusMetricsProvider
+    PROMETHEUS_METRICS_PROVIDER = NOOP_METRICS_PROVIDER
```
Comment (roman-anna-money) [inline aio_request/__init__.py:129]: А это что и для кого?))

Comment (Pliner) [reply aio_request/__init__.py:129]: For the backward compatibility. I don't want to break existing clients.

**Review by roman-anna-money — APPROVED** [review]

File: aio_request/client.py [hunk 2]
```diff
@@ -74,18 +89,122 @@ async def _request(
...
+    @contextlib.contextmanager
+    def _start_span(self, request: Request, started_at: int) -> Iterator[opentelemetry.trace.Span]:
+        if self._tracer is None:
+            self._tracer = opentelemetry.trace.get_tracer(__package__)
+
+        with self._tracer.start_as_current_span(
+            name=f"{request.method} {request.url}",
+            kind=opentelemetry.trace.SpanKind.CLIENT,
+            attributes={
+                opentelemetry.semconv.trace.SpanAttributes.HTTP_HOST: request.method,
```
Comment (spaceorc) [inline aio_request/client.py:179]: HOST это method???

**Review by spaceorc — APPROVED** [review]

File: aio_request/client.py (hunk 2 above)
Comment (Pliner) [reply aio_request/client.py]: Typo, sorry

**Pull request merged**